module-system/module-implementations/sov-attester-incentives/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+/// Call methods for the module
+pub mod call;
+
+/// Methods used to instantiate the module
+pub mod genesis;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "native")]
+#[allow(missing_docs)]
+pub mod query;
+
+use std::marker::PhantomData;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use call::Role;
+use sov_modules_api::{Context, Error};
+use sov_modules_macros::ModuleInfo;
+use sov_rollup_interface::zk::{
+    StoredCodeCommitment, ValidityCondition, ValidityConditionChecker, Zkvm,
+};
+use sov_state::{Storage, WorkingSet};
+
+/// Configuration of the attester incentives module
+pub struct AttesterIncentivesConfig<
+    C: Context,
+    Vm: Zkvm,
+    Cond: ValidityCondition,
+    Checker: ValidityConditionChecker<Cond>,
+> {
+    /// The address of the token to be used for bonding.
+    pub bonding_token_address: C::Address,
+    /// The address of the account holding the reward token supply
+    pub reward_token_supply_address: C::Address,
+    /// The minimum bond for an attester.
+    pub minimum_attester_bond: u64,
+    /// The minimum bond for a challenger.
+    pub minimum_challenger_bond: u64,
+    /// A code commitment to be used for verifying proofs
+    pub commitment_to_allowed_challenge_method: Vm::CodeCommitment,
+    /// A list of initial provers and their bonded amount.
+    pub initial_attesters: Vec<(C::Address, u64)>,
+    /// The finality period of the rollup (constant)
+    pub rollup_finality_period: u64,
+    /// The current maximum attested height
+    pub maximum_attested_height: u64,
+    /// The light client finalized height
+    pub light_client_finalized_height: u64,
+    /// The validity condition checker used to check validity conditions
+    pub validity_condition_checker: Checker,
```

Comment:
In general, you can't run the validity_condition_checker inside of the zkvm. These checks usually rely on either a trusted light client of the DA layer, or a ton of very SNARK unfriendly operations - so we'll need to think about how to handle them a little differently. 

I think the way to do it will be to add the validity_conditions as an argument to `apply_slot` so that they can get saved as part of the historical_transitions in the chain state module. That way, we can just do an equality check to ensure that the validity conditions from a challenge match the expected ones - and the actual checking of the conditions will get pushed to the node implementations. But that's way too much to try to do in this PR, so let's just leave this for now and open an issue.

Reply:
Makes a lot of sense. See my comment below for the issue